python/lsst/ip/diffim/dipoleFitTask.py

```diff
@@ -0,0 +1,921 @@
+                    doc="Dipole {0} lobe centroid".format(pos_neg))
+                setattr(self, ''.join((pos_neg, 'CentroidKey', x_y.upper())), key)
+
+        for x_y in ['x', 'y']:
+            key = schema.addField(
+                schema.join(name, "centroid", x_y), type=float, units="pixels",
+                doc="Dipole centroid")
+            setattr(self, ''.join(('centroidKey', x_y.upper())), key)
+
+        self.fluxKey = schema.addField(
+            schema.join(name, "flux"), type=float, units="dn",
+            doc="Dipole overall flux")
+
+        self.orientationKey = schema.addField(
+            schema.join(name, "orientation"), type=float, units="deg",
+            doc="Dipole orientation")
+
+        self.separationKey = schema.addField(
+            schema.join(name, "separation"), type=float, units="pixels",
+            doc="Pixel separation between positive and negative lobes of dipole")
+
+        self.chi2dofKey = schema.addField(
+            schema.join(name, "chi2dof"), type=float,
+            doc="Chi2 per degree of freedom of dipole fit")
+
+        self.signalToNoiseKey = schema.addField(
+            schema.join(name, "signalToNoise"), type=float,
+            doc="Estimated signal-to-noise of dipole fit")
+
+        self.classificationFlagKey = schema.addField(
+            schema.join(name, "flag", "classification"), type="Flag",
+            doc="Flag indicating diaSource is classified as a dipole")
+
+        self.flagKey = schema.addField(
+            schema.join(name, "flag"), type="Flag",
+            doc="General failure flag for dipole fit")
+
+        self.edgeFlagKey = schema.addField(
+            schema.join(name, "flag", "edge"), type="Flag",
+            doc="Flag set when dipole is too close to edge of image")
+
+    def measure(self, measRecord, exposure, posImage=None, negImage=None):
+        """Perform the non-linear least squares minimization.
+
+        The main functionality of this routine was placed outside of
+        this plugin (into `DipoleFitAlgorithm.fitDipole()`) so that
+        `DipoleFitAlgorithm.fitDipole()` can be called separately for
+        testing (see `tests/testDipoleFitter.py`)
+        """
+
+        pks = measRecord.getFootprint().getPeaks()
+        if len(pks) <= 1:  # not a dipole for our analysis
+            self.fail(measRecord, measBase.MeasurementError('not a dipole', self.FAILURE_NOT_DIPOLE))
+
+        result = None
+        try:
+            alg = DipoleFitAlgorithm(exposure, posImage=posImage, negImage=negImage)
+            result = alg.fitDipole(
+                measRecord, rel_weight=self.config.relWeight,
+                tol=self.config.tolerance,
+                maxSepInSigma=self.config.maxSeparation,
+                fitBgGradient=self.config.fitBgGradient,
+                separateNegParams=self.config.fitSeparateNegParams,
+                verbose=self.config.verbose, display=False)
```

Comment:
Good point. However, the current arrangement, which I prefer, is easier when I want to generate thousands of dipoles and test their fit (for evaluation, rather than basic testing).